profanity-im/profanity
# MUC MAM

MUC MAM

See #660

# How to test the functionality
* step 1: Have mam enabled
* step 2: Join random muc scroll up
* step 3: rejoining muc should fetch messages sent since last online with profanity

# I ran valgrind when using my new feature
no

----------------
# TODO
- [X] Look up older history
- [X] Look up history when creating muc window
- [X] Look up newer history
- [ ] Fetch mam on incoming message (does a new muc even get created on incoming message ?)
- [ ] Fetch mam on reconnects
- [x] Scroll history even without mam
- [ ] Why loading message sometimes lingers? (when at the end of muc mam maybe?)
- [X] Why is a chatwin opened for each member of muc?
- [ ] Last messages appear doubled (maybe this only happens for chatwins?)
- [X] wrong timestamp for messages through mam on new join of muc
- [x] mentions/triggers show printed date not original date
- [x] respect mam setting
- [X] get all rsm results even from pagination
- [ ] Autocompletion has to be correct
- [ ] how to handle read indicator with mam
- [ ] valgrind
- [ ] make loading mam for muc faster if possible
- [x] Don't request history from muc
